.goreleaser.yml

```diff
@@ -0,0 +1,68 @@
+before:
+  hooks:
+    - mkdir -p caddy-build
+    - cp cmd/caddy/main.go caddy-build/main.go
+    - cp ./go.mod caddy-build/go.mod
+    - sed -i.bkp s/github.com\/caddyserver\/caddy\/v2/caddy/g ./caddy-build/go.mod
+    # GoReleaser doesn't seem to offer {{.Tag}} at this stage, so we have to embed it into the env
+    # so we run: TAG=$(git describe --abbrev=0) goreleaser release --rm-dist --skip-publish --skip-validate
+    - go mod edit -require=github.com/caddyserver/caddy/v2@{{.Env.TAG}} ./caddy-build/go.mod
+    - git clone --depth 1 https://github.com/caddyserver/dist caddy-dist
+    - go mod download
+builds:
+- env:
+  - CGO_ENABLED=0
+  - GO111MODULE=on
+  main: main.go
+  dir: ./caddy-build
+  binary: caddy
+  goos:
+  - darwin
+  - linux
+  - windows
+  - freebsd
+  flags:
+  - -trimpath
+  ldflags:
+  - -s -w
+archives:
+- replacements:
+    darwin: Darwin
+    linux: Linux
+    windows: Windows
+    386: i386
+    amd64: x86_64
+checksum:
+  algorithm: sha256
+# signs:
+# - artifacts: all
+#   cmd: signify
+#   args: ["-S", "-s", "../seckey", "-m", "${artifact}"]
+nfpms:
+- package_name: caddy
+  homepage: https://caddyserver.com
+  description: Caddy Web Server
+  license: Apache 2.0
+  bindir: /usr/bin
+  formats:
+  - deb
+  - rpm
+  empty_folders:
+  - /etc/caddy
+  files:
```

Comment:
The `pre-install` and `post-install` scripts for the `deb` and `rpm` also need to be decided. I'm thinking we need to break the `packer/scripts/20-caddy.sh` file from the dist repo into 2 scripts: one to create the user and prep the directories, and the other to download the Caddy binary. Reuse the former script as one of the scripts to include as part of the packaging.

Reply:
I like reusing code :+1:

Reply:
It seems like managing {pre,post}{inst,rm} scripts is a [very tedious](https://wiki.debian.org/MaintainerScripts) [matter](https://www.debian.org/doc/debian-policy/ch-maintainerscripts.html). It doesn't seem to be as simple as packaging our scripts as-is. Let's put off producing the rpm/deb packages for now until we learn this better. Is that fine?

For posterity: https://www.debian.org/doc/manuals/maint-guide/index.en.html

Reply:
Yuck. That's a pain. Alright.